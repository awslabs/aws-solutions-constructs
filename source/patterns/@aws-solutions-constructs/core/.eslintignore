index.js
lib/*.js
test/*.js
*.d.ts
coverage
test/integ.*.js.snapshot/
test/cdk-integ.out.integ.*.snapshot
test/lambda/index.js
test/lambda-test/index.js