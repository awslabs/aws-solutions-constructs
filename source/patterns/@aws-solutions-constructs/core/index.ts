/**
 *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/alb-defaults';
export * from './lib/alb-helper';
export * from './lib/apigateway-defaults';
export * from './lib/apigateway-helper';
export * from './lib/dynamodb-table-defaults';
export * from './lib/fargate-defaults';
export * from './lib/fargate-helper';
export * from './lib/iot-topic-rule-defaults';
export * from './lib/kinesis-analytics-defaults';
export * from './lib/kinesis-analytics-helper';
export * from './lib/kinesis-streams-defaults';
export * from './lib/kinesis-streams-helper';
export * from './lib/kinesis-firehose-s3-defaults';
export * from './lib/lambda-event-source-mapping-defaults';
export * from './lib/kms-defaults';
export * from './lib/kms-helper';
export * from './lib/lambda-defaults';
export * from './lib/lambda-helper';
export * from './lib/mediastore-defaults';
export * from './lib/mediastore-helper';
export * from './lib/s3-bucket-defaults';
export * from './lib/s3-bucket-helper';
export * from './lib/secretsmanager-defaults';
export * from './lib/secretsmanager-helper';
export * from './lib/security-group-helper';
export * from './lib/sns-defaults';
export * from './lib/sns-helper';
export * from './lib/sqs-defaults';
export * from './lib/sqs-helper';
export * from './lib/cloudfront-distribution-helper';
export * from './lib/cloudfront-distribution-defaults';
export * from './lib/utils';
export * from './lib/events-rule-defaults';
export * from './lib/cognito-defaults';
export * from './lib/cognito-helper';
export * from './lib/elasticsearch-defaults';
export * from './lib/elasticsearch-helper';
export * from './lib/step-function-defaults';
export * from './lib/step-function-helper';
export * from './lib/dynamodb-table-helper';
export * from './lib/vpc-defaults';
export * from './lib/vpc-helper';
export * from './lib/sagemaker-defaults';
export * from './lib/sagemaker-helper';
export * from './lib/cloudwatch-log-group-helper';
export * from './lib/glue-job-defaults';
export * from './lib/glue-job-helper';
export * from './lib/glue-table-defaults';
export * from './lib/glue-table-helper';
export * from './lib/glue-database-defaults';
export * from './lib/glue-database-helper';
export * from './lib/input-validation';
export * from './test/test-helper';
export * from './lib/ssm-string-parameter-helper';
export * from './lib/eventbridge-helper';
export * from './lib/waf-defaults';
export * from './lib/waf-helper';