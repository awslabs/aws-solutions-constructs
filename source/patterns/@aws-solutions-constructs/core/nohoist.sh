packagess="
deep-diff
deepmerge
npmlog
"

mkdir node_modules
for package in $packagess; do
  echo "Installing $package"
  cp -r ../../../node_modules/$package ./node_modules/
done
