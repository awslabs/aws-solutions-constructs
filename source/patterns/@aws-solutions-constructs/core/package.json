{
  "name": "@aws-solutions-constructs/core",
  "version": "1.61.0",
  "description": "Core CDK Construct for patterns library",
  "main": "index.js",
  "types": "index.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/awslabs/aws-solutions-constructs.git",
    "directory": "source/patterns/@aws-solutions-constructs/core"
  },
  "author": {
    "name": "Amazon Web Services",
    "url": "https://aws.amazon.com",
    "organization": true
  },
  "license": "Apache-2.0",
  "scripts": {
    "build": "tsc -b .",
    "lint": "eslint -c ../eslintrc.yml --ext=.js,.ts . && tslint --project .",
    "lint-fix": "eslint -c ../eslintrc.yml --ext=.js,.ts --fix .",
    "test": "jest --coverage",
    "clean": "tsc -b --clean",
    "watch": "tsc -b -w",
    "integ": "cdk-integ",
    "integ-assert": "cdk-integ-assert",
    "jsii": "jsii",
    "jsii-pacmak": "jsii-pacmak",
    "build+lint+test": "npm run jsii && npm run lint && npm test && npm run integ-assert",
    "snapshot-update": "npm test -- -u"
  },
  "jsii": {
    "outdir": "dist",
    "targets": {
      "java": {
        "package": "software.amazon.awsconstructs.services.core",
        "maven": {
          "groupId": "software.amazon.awsconstructs",
          "artifactId": "core"
        }
      },
      "dotnet": {
        "namespace": "Amazon.Constructs",
        "packageId": "Amazon.Constructs",
        "signAssembly": true,
        "iconUrl": "https://raw.githubusercontent.com/aws/aws-cdk/master/logo/default-256-dark.png"
      },
      "python": {
        "distName": "aws-solutions-constructs.core",
        "module": "aws_solutions_constructs.core"
      }
    }
  },
  "dependencies": {
    "@aws-cdk/aws-cloudfront": "~1.61.0",
    "@aws-cdk/aws-cloudfront-origins": "~1.61.0",
    "@aws-cdk/aws-dynamodb": "~1.61.0",
    "@aws-cdk/aws-iot": "~1.61.0",
    "@aws-cdk/aws-kinesis": "~1.61.0",
    "@aws-cdk/aws-kinesisanalytics": "~1.61.0",
    "@aws-cdk/aws-kinesisfirehose": "~1.61.0",
    "@aws-cdk/aws-lambda": "~1.61.0",
    "@aws-cdk/aws-lambda-event-sources": "~1.61.0",
    "@aws-cdk/aws-logs": "~1.61.0",
    "@aws-cdk/aws-s3": "~1.61.0",
    "@aws-cdk/aws-sns": "~1.61.0",
    "@aws-cdk/aws-sqs": "~1.61.0",
    "@aws-cdk/core": "~1.61.0",
    "@aws-cdk/aws-iam": "~1.61.0",
    "@aws-cdk/aws-apigateway": "~1.61.0",
    "@aws-cdk/aws-kms": "~1.61.0",
    "@aws-cdk/aws-events": "~1.61.0",
    "@aws-cdk/aws-cognito": "~1.61.0",
    "@aws-cdk/aws-elasticsearch": "~1.61.0",
    "@aws-cdk/aws-cloudwatch": "~1.61.0",
    "@aws-cdk/aws-stepfunctions": "~1.61.0",
    "@types/deep-diff": "^1.0.0",
    "@types/npmlog": "^4.1.2",
    "deep-diff": "^1.0.2",
    "deepmerge": "^4.0.0",
    "npmlog": "^4.1.2"
  },
  "devDependencies": {
    "@aws-cdk/assert": "~1.61.0",
    "@types/jest": "^24.0.23",
    "@types/node": "^10.3.0"
  },
  "jest": {
    "moduleFileExtensions": [
      "js"
    ]
  },
  "bundledDependencies": [
    "deepmerge",
    "npmlog",
    "@types/npmlog",
    "deep-diff",
    "@types/deep-diff"
  ],
  "peerDependencies": {
    "@aws-cdk/aws-cloudfront": "~1.61.0",
    "@aws-cdk/aws-dynamodb": "~1.61.0",
    "@aws-cdk/aws-iot": "~1.61.0",
    "@aws-cdk/aws-kinesis": "~1.61.0",
    "@aws-cdk/aws-kinesisanalytics": "~1.61.0",
    "@aws-cdk/aws-kinesisfirehose": "~1.61.0",
    "@aws-cdk/aws-lambda": "~1.61.0",
    "@aws-cdk/aws-lambda-event-sources": "~1.61.0",
    "@aws-cdk/aws-logs": "~1.61.0",
    "@aws-cdk/aws-s3": "~1.61.0",
    "@aws-cdk/aws-sns": "~1.61.0",
    "@aws-cdk/aws-sqs": "~1.61.0",
    "@aws-cdk/core": "~1.61.0",
    "@aws-cdk/aws-iam": "~1.61.0",
    "@aws-cdk/aws-apigateway": "~1.61.0",
    "@aws-cdk/aws-kms": "~1.61.0",
    "@aws-cdk/aws-events": "~1.61.0",
    "@aws-cdk/aws-cognito": "~1.61.0",
    "@aws-cdk/aws-elasticsearch": "~1.61.0",
    "@aws-cdk/aws-cloudwatch": "~1.61.0",
    "@aws-cdk/aws-stepfunctions": "~1.61.0",
    "@aws-cdk/aws-cloudfront-origins": "~1.61.0"
  }
}
