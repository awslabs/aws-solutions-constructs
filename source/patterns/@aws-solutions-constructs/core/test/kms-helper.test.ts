/**
 *  Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

// Imports
import { Stack } from "@aws-cdk/core";
import * as defaults from '../';
import { SynthUtils, ResourcePart } from '@aws-cdk/assert';
import '@aws-cdk/assert/jest';

// --------------------------------------------------------------
// Test minimal deployment with no properties
// --------------------------------------------------------------
test('Test minimal deployment with no properties', () => {
    // Stack
    const stack = new Stack();
    // Helper declaration
    defaults.buildEncryptionKey(stack);
    // Assertion 1
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    // Assertion 2
    expect(stack).toHaveResourceLike('AWS::KMS::Key', {
        Type: "AWS::KMS::Key",
        Properties: {
            EnableKeyRotation: true
        }
      }, ResourcePart.CompleteDefinition);
});

// --------------------------------------------------------------
// Test deployment w/ custom properties
// --------------------------------------------------------------
test('Test minimal deployment with custom properties', () => {
  // Stack
  const stack = new Stack();
  // Helper declaration
  defaults.buildEncryptionKey(stack, {
    enableKeyRotation: false
  });
  // Assertion 1
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  // Assertion 2
  expect(stack).toHaveResourceLike('AWS::KMS::Key', {
      Type: "AWS::KMS::Key",
      Properties: {
          EnableKeyRotation: false
      }
    }, ResourcePart.CompleteDefinition);
});