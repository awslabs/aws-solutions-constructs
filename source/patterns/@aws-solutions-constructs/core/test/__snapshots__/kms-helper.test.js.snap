// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test minimal deployment with custom properties 1`] = `
Object {
  "Resources": Object {
    "EncryptionKey1B843E66": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "EnableKeyRotation": false,
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
                "kms:TagResource",
                "kms:UntagResource",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;

exports[`Test minimal deployment with no properties 1`] = `
Object {
  "Resources": Object {
    "EncryptionKey1B843E66": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "EnableKeyRotation": true,
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
                "kms:TagResource",
                "kms:UntagResource",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
